### PR TITLE
Add gitops upgrade doc

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

#2755 

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/tasks/cluster/cluster-upgrades.md [hunk 1]
```diff
@@ -168,6 +168,7 @@ allowing you to upgrade a number of fields simultaneously with the same procedur
 - `workerNodeGroupConfigurations.machineGroupRef.name`
 - `etcdConfiguration.externalConfiguration.machineGroupRef.name`
 - `identityProviderRefs` (Only for `kind:OIDCConfig`, `kind:AWSIamConfig` is immutable)
+- `gitOpsRef` (Only support adding this field from not defined)
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/cluster/cluster-upgrades.md:171]:
```suggestion
- `gitOpsRef` (Once set, you can't change or delete the field's content later)
```

**Review by chrisnegus — COMMENTED** [review]
I suggested some other wording. Otherwise, lgtm.

**Review by abhinavmpandey08 — APPROVED** [review]

**Pull request merged**